### PR TITLE
Refactor DMLQueryBuilder

### PR DESCRIPTION
**Pull request by Tigrov**
Related with yiisoft/db#746

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

File: src/DMLQueryBuilder.php [hunk 1]
```diff
@@ -74,42 +69,28 @@ public function upsert(
...
-            $updateColumns,
-        );
+        [$uniqueNames, , $updateNames] = $this->prepareUpsertColumns($table, $insertColumns, $updateColumns);
 
         if (empty($uniqueNames)) {
             return $insertSql;
         }
 
-        if ($updateNames === []) {
+        if ($updateColumns === false || $updateNames === []) {
```
Comment (vjik) [inline src/DMLQueryBuilder.php:78]:
```suggestion
        if (empty($updateColumns)) {
```

It's a little more productive.

File: src/DMLQueryBuilder.php [hunk 2]
```diff
@@ -74,42 +69,28 @@ public function upsert(
...
-
-        if ($updateColumns === false) {
             return "$insertSql ON CONFLICT DO NOTHING";
         }
 
         if ($updateColumns === true) {
             $updateColumns = [];
 
-            /** @psalm-var string $name */
+            /** @psalm-var string[] $updateNames */
```
Comment (vjik) [inline src/DMLQueryBuilder.php:86]:
```suggestion
```

Seems, it not need after last changes in PR to yiisoft/db.

**Review by vjik — APPROVED** [review]

File: src/DMLQueryBuilder.php (hunk 1 above)
Comment (Tigrov) [reply src/DMLQueryBuilder.php:78]: Here is not only `$updateColumns === false` but also `$updateNames === []`

and `$updateNames` could be `null`

File: src/DMLQueryBuilder.php (hunk 2 above)
Comment (Tigrov) [reply src/DMLQueryBuilder.php:86]: `$updateNames` could be `null` but if `$updateColumns === true` it is `array`

https://github.com/yiisoft/db/pull/746/files#diff-697d8c4d18282e94c125fbb9302841ce34c6f6bbe6a4406cd57b371ca9e145aeR299-R303

**Pull request merged**